File: Dataset/Leetcode/test/98/550.java
 class Solution {
    long pre = Long.MIN_VALUE;
    public boolean XXX(TreeNode root) {
        if (root == null)
            return true;
        if (root.left != null){
            if (!XXX(root.left))
                return false;
        }
        if (root.val <= pre)
            return false;
        else
            pre = root.val;
        if (root.right != null)
            return XXX(root.right);
        else 
            return true;
    }
}

File: Dataset/Leetcode/valid/28/550.cpp
 class Solution {
public:
    int XXX(string haystack, string needle) 
    {
        int l1 = haystack.size(), l2 = needle.size();
        if(l2 == 0) return 0;

        if(l1 < l2) return -1;

        for(int i = 0; i < l1; i++)
        {
            if(haystack[i] == needle[0])
            {
                for(int j = 0; j < l2; j++)
                {
                    if(i+j >= l1 || haystack[i+j] != needle[j])
                    {
                        break;
                    }
                    if(j == l2-1) return i;
                }
            }
        }

        return -1;
    }
};

